modules/apps/portal-search/portal-search-rest-test/src/testIntegration/java/com/liferay/portal/search/rest/resource/v1_0/test/SearchResultResourceTest.java

```diff
@@ -1156,54 +1193,49 @@ private void _testPostSearchPageWithKeywords() throws Exception {
 	private void _testPostSearchPageWithLocalizedTextObjectField()
 		throws Exception {
 
-		ObjectDefinition objectDefinition =
-			_objectDefinitionLocalService.addCustomObjectDefinition(
-				_user.getUserId(), 0, null, false, true, true, false,
-				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
-				ObjectDefinitionTestUtil.getRandomName(), null, null,
-				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
-				true, ObjectDefinitionConstants.SCOPE_COMPANY,
-				ObjectDefinitionConstants.STORAGE_TYPE_DEFAULT,
-				Arrays.asList(
-					new TextObjectFieldBuilder(
-					).labelMap(
-						LocalizedMapUtil.getLocalizedMap(
-							RandomTestUtil.randomString())
-					).indexed(
-						true
-					).name(
-						"localizedTextObjectFieldName"
-					).localized(
-						true
-					).build()));
-
-		objectDefinition =
-			_objectDefinitionLocalService.publishCustomObjectDefinition(
-				_user.getUserId(), objectDefinition.getObjectDefinitionId());
-
 		DTOConverterContext dtoConverterContext =
 			new DefaultDTOConverterContext(
 				false, Collections.emptyMap(), _dtoConverterRegistry, null,
 				LocaleUtil.getDefault(), null, TestPropsValues.getUser());
 
 		_objectEntryManager.addObjectEntry(
-			dtoConverterContext, objectDefinition,
+			dtoConverterContext, _objectDefinition,
+			new ObjectEntry() {
+				{
+					properties = HashMapBuilder.<String, Object>put(
+						"localizedTextObjectFieldName_i18n",
+						HashMapBuilder.put(
+							"en_US", "paul"
+						).put(
+							"pt_BR", "paulo"
+						).build()
+					).build();
+				}
+			},
+			ObjectDefinitionConstants.SCOPE_COMPANY);
+
+		_objectEntryManager.addObjectEntry(
+			dtoConverterContext, _objectDefinition,
 			new ObjectEntry() {
 				{
 					properties = HashMapBuilder.<String, Object>put(
 						"localizedTextObjectFieldName_i18n",
 						HashMapBuilder.put(
-							"en_US", "en_US localizedTextObjectFieldValue"
+							"en_US", "peter"
 						).put(
-							"pt_BR", "pt_BR localizedTextObjectFieldValue"
+							"pt_BR", "pedro"
 						).build()
 					).build();
 				}
 			},
 			ObjectDefinitionConstants.SCOPE_COMPANY);
 
+		PermissionThreadLocal.setPermissionChecker(
+			PermissionCheckerFactoryUtil.create(TestPropsValues.getUser()));
+		PrincipalThreadLocal.setName(TestPropsValues.getUserId());
```

Comment:
@vngoliveira please use try/catch:

```
PermissionChecker originalPermissionChecker =
	PermissionThreadLocal.getPermissionChecker();
String originalName = PrincipalThreadLocal.getName();

try {
	PermissionThreadLocal.setPermissionChecker(
		PermissionCheckerFactoryUtil.create(TestPropsValues.getUser()));
	PrincipalThreadLocal.setName(TestPropsValues.getUserId());

	....
}
finally {
	PermissionThreadLocal.setPermissionChecker(
		originalPermissionChecker);
	PrincipalThreadLocal.setName(originalName);
}

```

Reply:
Sorry for the confusion, I forgot to delete this code!  
Changing the object deletion way is sufficient for me.